lib/utils/deeply-non-duplicated-addon.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+/**
+ * Deduplicate one addon's children addons recursively from hostAddons.
+ *
+ * @private
+ * @param {Object} hostAddons
+ * @param {EmberAddon} dedupedAddon
+ * @param {String} treeName
+ */
+module.exports = function deeplyNonDuplicatedAddon(hostAddons, dedupedAddon, treeName) {
+  if (dedupedAddon.addons.length === 0) {
+    return;
+  }
+
+  dedupedAddon._orginalAddons = dedupedAddon.addons;
+  dedupedAddon.addons = dedupedAddon.addons.filter(addon => {
+    if (addon.lazyLoading && addon.lazyLoading.enabled) {
```

Comment:
you need check if `addon.lazyLoading` is a bollean, please

Reply:
we should be able to reuse `ensureLazyLoadingHash` from #610

Reply:
I think https://github.com/ember-engines/ember-engines/blob/master/lib/engine-addon.js#L384-L390 already makes sure that `addon.lazyLoading` is a hash. Do we still have to use `ensureLazyLoadingHash` each time before we use `addon.lazyLoading`?

@villander Have you got a issue if you don't have [#610](https://github.com/ember-engines/ember-engines/pull/610)?

Reply:
@2hu12 What Dan wanted to say is that we now have a utility and it makes more sense to keep the code standardized from now. 

And in your code too, if the check in hook init (https://github.com/ember-engines/ember-engines/blob/master/lib/engine-addon.js#L384-L390) already solves your problem why do the "if" here again?

Reply:
@villander Thank you for the explanation.

The `if` here is to make sure that there's a `lazyLoading` option and the `enabled` property is `true`, if one engine has no `lazyLoading` at all using `if (addon.lazyLoading.enabled)` would cause runtime error.

And my question is should the `ensureLazyLoadingHash` helper be used once in the `init` hook or every time before it reads the `addon.lazyLoading.enabled`?

Reply:
@2hu12 In your case, it should be used in hook init and in your solution because you are doing a scan on all addons/dependencies

try to use only in init and see if something breaks

Reply:
@villander Could you please give me a example of how to test if something breaks? like running `npm run test`?

Reply:
@2hu12 can you finish this to be able to finish?

I would love see this merged

Reply:
@villander I just did some manual tests, the dummy app which has an engine called `ember-blog` with `lazyLoading: true` seems ok with the latest change and all tests pass locally.